libraries/Microsoft.Bot.Builder.Runtime/Providers/IProvider.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Bot.Builder.Runtime.Providers
+{
+    public interface IProvider
+    {
+        void ConfigureServices(IServiceCollection services, IConfiguration configuration);
```

Comment:
The need for this is not clear.

Reply:
Documentation added in latest commit. The purpose of this class is to ultimately provide a pattern in which we can encapsulate the registration of services for DI in such a way that they can be serialized with JSON.